src/backend/distributed/metadata/dependency.c

```diff
@@ -1022,13 +1022,14 @@ GetUndistributableDependency(const ObjectAddress *objectAddress)
 		if (!SupportedDependencyByCitus(dependency))
 		{
 			/*
-			 * Skip roles and text search templates.
+			 * Skip roles, access methods, and text search templates.
```

Comment:
I think our commit message for this change is hard to digest.

Now, PG records the AM entries properly after SET ACCESS METHOD, which is fine & useful.

But it is not very clear why this code has changed here (e.g., GetUndistributableDependency), maybe because we still have references to "enterprise" in the code which distracts me. 

I guess (a) it is not possible to distribute AMs via object propagation (b) now that AMs are recorded properly, without this change, we'd try to distribute AMs, which would fail? 

Would be nice to improve the comment. Also, do we already have a test for that?

Reply:
Sounds good, I will improve the comment.
Yes, we have a test for this, it's the one that was failing before this change:
https://github.com/citusdata/citus/blob/main/src/test/regress/sql/pg15.sql#L930

Before this change, it would fail like this:
```diff
 ALTER TABLE mx_ddl_table2 SET ACCESS METHOD heap2;
+ERROR:  "table mx_ddl_table2" has dependency on unsupported object "access method heap2"
```

Reply:
Let's check the full test for this:
```SQL
-- SET ACCESS METHOD
-- Create a heap2 table am handler with heapam handler
CREATE ACCESS METHOD heap2 TYPE TABLE HANDLER heap_tableam_handler;
SELECT run_command_on_workers($$CREATE ACCESS METHOD heap2 TYPE TABLE HANDLER heap_tableam_handler$$);
CREATE TABLE mx_ddl_table2 (
    key int primary key,
    value int
);
SELECT create_distributed_table('mx_ddl_table2', 'key', 'hash', shard_count=> 1);

ALTER TABLE mx_ddl_table2 SET ACCESS METHOD heap2;
NOTICE:  issuing ALTER TABLE mx_ddl_table2 SET ACCESS METHOD heap2;
NOTICE:  issuing SELECT worker_apply_shard_ddl_command (102008, 'public', 'ALTER TABLE mx_ddl_table2 SET ACCESS METHOD heap2;')
```
First the test manually creates the AM in the workers as well through run_command_on_workers. Then when we run the ALTER command, it's simply propagated as it is to the shards. Note that in 15.3 if the AM hadn't been manually created on the workers, the ALTER command would fail with a `not exists` error.

To preserve the same behavior now that AM is recorded as a dependency in 15.4, I ignore the AM dependency by saying in the comment that the user is responsible for the AM to exist in the workers, as it was before as well.